src/syntaxtree.c

```diff
@@ -199,6 +295,22 @@ static Obj SyntaxTreeRangeExpr(Obj result, Expr expr)
     return result;
 }
 
+static Expr SyntaxTreeCodeRangeExpr(Obj node)
+{
+    UInt hassecond = IsbPRec(node, RNamName("second"));
```

Comment:
To get "free" input validation, use this:
```suggestion
    UInt hassecond = ISB_REC(node, RNamName("second"));
```
Of course on the long run, slightly more helpful errors would be nice (say, saying "Expected <input>.statements.stats.foo.bar` to be a record, not a %s"), but implementing this would take substantial effort, which at this stage seems premature.